Sources/MapboxCoreNavigation/Locale.swift

```diff
@@ -23,20 +23,13 @@ extension Locale {
      Returns a `Locale` from `preferredLocalLanguageCountryCode`.
      */
     public static var nationalizedCurrent = Locale(identifier: preferredLocalLanguageCountryCode)
-    
+
     public static var usesMetric: Bool {
-        let locale = self.current as NSLocale
-        guard let measurementSystem = locale.object(forKey: .measurementSystem) as? String else {
-            return false
-        }
-        return measurementSystem == "Metric"
+        return current.usesMetricSystem
     }
     
+    @available(*, deprecated, renamed: "usesMetricSystem")
     public var usesMetric: Bool {
```

Comment:
`usesMetricSystem` behaves differently for `en-GB` locale, because the UK uses miles and yards for road distances and feet and inches for width and height restrictions, but they use the metric system for most other purposes: https://github.com/mapbox/mapbox-navigation-ios/pull/383#discussion_r127935522.

```swift
Locale(identifier: "en-GB").usesMetricSystem // true
NSLocale(localeIdentifier: "en-GB").usesMetricSystem // true
NSLocale(localeIdentifier: "en-GB").object(forKey: .measurementSystem) // "U.K."
```

This method is specifically used to decide the units for distances:

https://github.com/mapbox/mapbox-navigation-ios/blob/db95f8b68b528027036a92a83eb425274b88a905/Sources/MapboxNavigation/NavigationViewController.swift#L218

Instead of deprecating this method, we should rename it to `measuresDistancesInMeters`. Maybe there’s something in `MeasurementFormatter` that can make this decision more data-driven and avoid the use of NSLocale objects.

Reply:
That makes sense, thanks, I will revisit this PR

Reply:
Renamed